scrape_bonds.py

```
@@ -0,0 +1,298 @@
+            el = page.locator(sel).first
+            if el and el.count() > 0:
+                txt = el.inner_text(timeout=2000).strip()
+                # Extract first number
+                m = re.search(r'(\d{1,3}(?:,\d{3})*(?:\.\d+)?|\d+(?:\.\d+)?)', txt.replace(',', ''))
+                if m:
+                    price_val = float(m.group(1))
+                    break
+        except PWTimeout:
+            continue
+        except Exception:
+            continue
```

Ruff (0.13.3)

`try`-`except`-`continue` detected, consider logging the exception

(S112)